### PR TITLE
Adding paramater for stringsAsFactors option

### PR DESCRIPTION
**Pull request by geneorama**
Updated tests to check column classes and return type
 Converted calls to 'content' to the fully qualified name 'httr::content'
 Updated DESCRIPTION
 Updated documentation for new parameter

### CODE REVIEW

File: tests/testthat/test-all.R [hunk 1]
```diff
@@ -85,18 +85,56 @@ test_that("Fields with currency symbols remove the symbol and convert to money",
...
+})
+
+
+test_that("read Socrata JSON as default / factors", {
   df <- read.socrata('https://soda.demo.socrata.com/resource/4334-bgaj.json')
+  expect_equal("data.frame", class(df), label="class")
   expect_equal(1007, nrow(df), label="rows")
   expect_equal(11, ncol(df), label="columns")
+  expect_equal(c("factor", "factor", "factor", "factor", "factor", "factor", 
+                 "factor", "factor", "factor", "factor", "factor"),
```
Comment (geneorama) [inline tests/testthat/test-all.R]: Some of these should be POSIX #85

File: tests/testthat/test-all.R [hunk 2]
```diff
@@ -85,18 +85,56 @@ test_that("Fields with currency symbols remove the symbol and convert to money",
...
 
+test_that("read Socrata JSON as character", {
+  df <- read.socrata('https://soda.demo.socrata.com/resource/4334-bgaj.json',
+                     stringsAsFactors = FALSE)
+  expect_equal("data.frame", class(df), label="class")
+  expect_equal(1007, nrow(df), label="rows")
+  expect_equal(11, ncol(df), label="columns")
+  expect_equal(c("character", "character", "character", "character", "character", 
+                 "character", "character", "character", "character", "character", 
+                 "character"),
```
Comment (geneorama) [inline tests/testthat/test-all.R]: Some of these should be POSIX #85

**Comment by tomschenkjr** [comment]
@geneorama - I found the issue and, yes, it is linked to your code changes.

The unit test on replacing the data set is written as:

1. Come up with a two columns of random numbers
2. Upload data set to portal using `write.socrata()`
3. Download the data you just uploaded and make sure the data matches

In step (3), that changed with your code changes. Namely, the data downloaded from the portal is being interpreted as factors, even though it's numeric.

So, you didn't break `write.socrata()`, but how `read.socrata()` is downloading data did change, which causes the test to fail.

When you step through the `fully replace a dataset` test, you can easily pick out the errors. Please update the pull request.

**Comment by geneorama** [comment]
@tomschenkjr Good catch, I didn't realize that there was a `read.socrata` dependency in the write operation.

So, the issue is a little more nuanced; for consistency I ensure that all character output (JSON and CSV) is downloaded as a factor.  However, we have an issue with JSON downloads where all the fields are character (including numeric) so they are now factor, and conversion from factor with `as.numeric` returns the factor level rather than the number.  

One way to fix this would be to first convert the factor back to character in the test, then convert it to numeric.

However, I suggest that we fix this another way; return all strings as character.

**Comment by geneorama** [comment]
I went ahead and changed the default and the tests to get the tests working (hopefully).  I can revert if you don't want to go that route.

**Comment by tomschenkjr** [comment]
It's a good placeholder for now. Looks like the tests passed.


Tom Schenk Jr.

Chief Data Officer

Department of Innovation and Technology

City of Chicago

(312) 744-2770

tom.schenk@cityofchicago.org

data.cityofchicago.org


________________________________

**Comment by tomschenkjr** [comment]
@geneorama - this looks good. Thanks for the test.

**Pull request merged**